lib/rules/no-jquery-methods.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * @fileOverview Disallow the use of specific jQuery methods.
+ * @author Marc Lynch
+ */
+'use strict';
+
+/**
+ * Determines if this expression matches the method blacklist invocation.
+ * @param {ASTNode} node The identifier node.
+ * @returns {Boolean} Returns true if the expression matches, otherwise false.
+ */
+
+function get(node, path) {
+  let parts;
+  if (typeof path === 'string') {
+    parts = path.split('.');
+  } else {
+    parts = path;
+  }
+
+  if (parts.length === 1) {
+    return node[path];
+  }
+
+  let property = node[parts[0]];
+
+  if (property && parts.length > 1) {
+    parts.shift();
+    return get(property, parts);
+  }
+
+  return property;
+}
+
+function getMessage(blackListName) {
+  return `The use of ${blackListName} method has been deprecated.`;
+}
+
+function isJQueryCaller(node, name = null) {
+  let oName = get(node, 'object.callee.object.name') === 'Ember';
+  let pName = get(node, 'object.callee.property.name') === '$';
+  let cName = get(node, 'object.callee.name') === '$';
+  let lName = get(node, 'object.name') === name;
+
+  return oName || pName || cName || lName;
+}
+
+function getLocalImportName(node, sourceName) {
+  if (get(node, 'source.value').includes(sourceName)) {
+    return get(node, 'specifiers').map((specifier) => specifier.local.name);
+  }
+
+  return [];
+}
+
+module.exports = {
+  docs: {
+    category: 'Best Practices',
+    recommended: true
+  },
+  meta: {
+    message: getMessage
+  },
+  create(context) {
+    const BLACKLIST = context.options[0] || [];
+    let jQueryLocalName = null;
+    let varDecIDName = null;
+
+    return {
+      ImportDeclaration(node) {
+        let localName = getLocalImportName(node, 'jquery');
+        if (localName.length > 0) {
+          jQueryLocalName = localName[0];
+        }
+      },
+      VariableDeclarator(node) {
```

Comment:
This is a bit of a nit, but it goes toward readability, but can we add a blank line in between these members?

```js
return {
  ImportDeclaration(node) {
    let localName = getLocalImportName(node, 'jquery');
    if (localName.length > 0) {
       jQueryLocalName = localName[0];
    }
  },

VariableDeclarator(node) {
...
```